### ISSUE TITLE
Large discrepancy with strtod() for certain numbers

### ISSUE THREAD

**Issue by EricBracken** [grzegorz-kraszewski/stringtofloat#1]
For the input string "8.757022884609e-12", it's only single-precision accurate for me:
```
  strtod =  8.75702288460900036e-12
 str2dbl =  8.75702285750394605e-12
    diff =  2.71050543121376109e-20
relative =  3.09523620861757650e-09
 strtod: 0011110110100011010000011100010000000000000000000000000000000000
str2dbl: 0011110110100011010000011100001111111111000000000000000000000000
```
The compiler is Visual Studio 2015.  Some other numbers that had similar problems were:
1.612387501293e-09
5.294512883669e-10
9.540519763096e-11
5.551220733091e-10
1.290643536489e-09
6.907961008773e-10
8.597235201524e-09
4.101435358006e-09
9.333679218404e-09
1.801570559401e-10

**Comment by grzegorz-kraszewski** [comment]
I have confirmed the problem using GCC 5.4.0 on x86_64, Linux. Will investigate more next weekend.

**Comment by grzegorz-kraszewski** [comment]
Step 1 of debugging, again reproduced the problem using GCC 11.4.0 on Linux, x86_64.

**Comment by grzegorz-kraszewski** [comment]
Step 2: result is exactly the same regardless of `USE_64BIT_FOR_ADDSUB_MACROS`, so my first assumption is, the bug is elsewhere. Adding step by step debug output.

Step 3: parser stage is good.

**Comment by grzegorz-kraszewski** [comment]
The bug is fixed and deserves an explanation. It has been located in the last stage of `converter()`, assembling final *double* number from parts.
```c
q = ((uint64_t)(s2 & ~mask28) << 24) | (((uint64_t)s1 + 128) >> 8) | binexs2;
```
Notice `+ 128` expression. This is a lame attempt to implement "round to nearest" rounding mode, which contradicts the statement, that `str2dbl()` works in "round towards zero" mode. But the main problem is that this "half of a least significant bit" may cause carry, which is **not propagated** to the upper part.

The fix is to simply remove this `+ 128`, then the code really works in "round towards zero" mode.

Results after fix:
```
gcc_converted: 8.757022884609000357e-12
s2f_converted: 8.757022884608998742e-12
absolute difference: 1.615587133892632177e-27
relative difference: 1.844904547105985949e-16
gcc_converted: 0011110110100011010000011100010000000000000000000000000000000000
s2f_converted: 0011110110100011010000011100001111111111111111111111111111111111
```
The difference is now one least significant bit and comes from different rounding modes (following bits of the mantissa are 11000111...).

Going to test the rest of testcases.

**Comment by grzegorz-kraszewski** [comment]
Results for the rest of testcases:
```
GCC converted           S2F converted           absolute diff           relative diff
1.6123875012930000e-09  1.6123875012929998e-09  2.0679515313825692e-25  1.2825400406070160e-16
5.2945128836690003e-10  5.2945128836689993e-10  1.0339757656912846e-25  1.9529195384160786e-16
9.5405197630960004e-11  9.5405197630959991e-11  1.2924697071141057e-26  1.3547162410517197e-16
5.5512207330910002e-10  5.5512207330909991e-10  1.0339757656912846e-25  1.8626097130811656e-16
1.2906435364890001e-09  1.2906435364889999e-09  2.0679515313825692e-25  1.6022638884537536e-16
6.9079610087730003e-10  6.9079610087729993e-10  1.0339757656912846e-25  1.4967886535233073e-16
8.5972352015240006e-09  8.5972352015239989e-09  1.6543612251060553e-24  1.9242944811057312e-16
4.1014353580060003e-09  4.1014353580059995e-09  8.2718061255302767e-25  2.0168076303784025e-16
9.3336792184040007e-09  9.3336792184039991e-09  1.6543612251060553e-24  1.7724641980881581e-16
1.8015705594010001e-10  1.8015705594009998e-10  2.5849394142282115e-26  1.4348255197330001e-16
```

**Comment by grzegorz-kraszewski** [comment]
Fixed in commit [d113745](https://github.com/grzegorz-kraszewski/stringtofloat/commit/d1137455dfcab03c3d7f2f58beec31219c5efb94).
